src/main/java/com/ericsson/ei/erqueryservice/ERQueryService.java

```diff
@@ -112,33 +114,33 @@ public void setRest(RestOperations rest) {
      *            whether or not to retain the tree structure in the result.
      * @return ResponseEntity
      */
-    public ResponseEntity<JsonNode> getEventStreamDataById(String eventId, SearchOption searchOption, int limit,
+    public ResponseEntity getEventStreamDataById(String eventId, SearchOption searchOption, int limit,
             int levels, boolean tree) {
 
-        if(StringUtils.isNotBlank(url)) {
-            final String erUrl = URI.create(url.trim() + "/" + eventId).normalize().toString();
-            LOGGER.debug("The URL to ER is: {}", erUrl);
-
-            // Request Body parameters
-            final SearchParameters searchParameters = getSearchParameters(searchOption);
-
-            // Build query parameters
-            final UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(erUrl).queryParam("limit", limit)
-                    .queryParam("levels", levels).queryParam("tree", tree);
-            final HttpHeaders headers = new HttpHeaders();
-            headers.setContentType(MediaType.APPLICATION_JSON);
-
-            final HttpEntity<SearchParameters> requestEntity = new HttpEntity<>(searchParameters, headers);
-            final UriComponents uriComponents = builder.buildAndExpand(searchParameters);
-            LOGGER.debug("The request is : {}", uriComponents.toUri().toString());
-
-            try {
-                return rest.exchange(uriComponents.toUri(), HttpMethod.POST, requestEntity, JsonNode.class);
-            } catch (RestClientException e) {
-                LOGGER.error("Error occurred while executing REST POST to: {} for\n{}", erUrl, requestEntity, e);
+        String uri = null;
+        try {
+            if(StringUtils.isNotBlank(erBaseUrl)) {
+                // Request Body parameters
+                final SearchParameters searchParameters = getSearchParameters(searchOption);
+                request
+                        .setHttpMethod(HttpMethod.POST)
+                        .setBaseUrl(erBaseUrl)
+                        .setEndpoint(eventId)
```

Comment:
isn't "events" entry point missing here?,, if I remember correct eventid value follows after the "events" entrypoint.

Reply:
I'm using the exactly same url as before, which works.

This is the old builup.
`final String erUrl = URI.create(url.trim() + "/" + eventId).normalize().toString();`

Reply:
The search endpoint is part of the er.url

Reply:
Ok,, maybe same comment as previous comment.